platform/core/src/test/kotlin/spp/platform/core/SourceServiceITTest.kt

```diff
@@ -18,18 +18,511 @@
+        Assertions.assertNull(addAccessPermissionResp.getJsonArray("errors"))
+        val accessPermission = addAccessPermissionResp.getJsonObject("data").getJsonObject("addAccessPermission")
+        Assertions.assertEquals(AccessType.WHITE_LIST.name, accessPermission.getString("type"))
+        Assertions.assertEquals("some-pattern", accessPermission.getJsonArray("locationPatterns")[0])
+
+        val accessPermissionId = accessPermission.getString("id")
+        val getAccessPermissionResp = request
+            .sendJsonObject(
+                JsonObject().put(
+                    "query",
+                    """query (${'$'}id: String!){
+                          getAccessPermission (id: ${'$'}id){
+                                id
+                                locationPatterns
+                                type
+                            }
+                        }
+                    """.trimIndent()
+                ).put("variables", JsonObject().put("id", accessPermissionId))
+            ).await().bodyAsJsonObject()
+        Assertions.assertNull(getAccessPermissionResp.getJsonArray("errors"))
+        val accessPermission2: JsonObject = getAccessPermissionResp.getJsonObject("data")
+            .getJsonObject("getAccessPermission")
+        Assertions.assertEquals(AccessType.WHITE_LIST.name, accessPermission2.getString("type"))
+        Assertions.assertEquals("some-pattern", accessPermission2.getJsonArray("locationPatterns")[0])
+    }
+
+
+    @Test
+    fun `ensure addDeveloper works`() = runBlocking {
+        val addDeveloperResp = request.sendJsonObject(addDeveloperRequest).await().bodyAsJsonObject()
+        Assertions.assertNull(addDeveloperResp.getJsonArray("errors"))
+        val developer: JsonObject = addDeveloperResp.getJsonObject("data").getJsonObject("addDeveloper")
+        Assertions.assertEquals("developer-id", developer.getString("id"))
+        Assertions.assertNotNull(developer.getString("accessToken"))
+    }
+
+
+    @Test
+    fun `ensure getDevelopers works`() = runBlocking {
+        val addDeveloperResp = request.sendJsonObject(addDeveloperRequest).await().bodyAsJsonObject()
+        Assertions.assertNull(addDeveloperResp.getJsonArray("errors"))
+
+        val getDevelopersResp = request
+            .sendJsonObject(
+                JsonObject().put(
+                    "query",
+                    """query {
+                          getDevelopers {
+                                id
+                                accessToken
+                            }
+                        }
+                    """.trimIndent()
+                )
+            ).await().bodyAsJsonObject()
+        Assertions.assertNull(getDevelopersResp.getJsonArray("errors"))
+        val developers = getDevelopersResp.getJsonObject("data").getJsonArray("getDevelopers")
+        Assertions.assertEquals("developer-id", developers.getJsonObject(0).getString("id"))
```

Comment:
I don't believe there is a guaranteeded order of `getDevelopers`. Looks like this test is failing because of that.